community/membership.md

```diff
@@ -0,0 +1,286 @@
+    - Authoring or reviewing PRs on GitHub
+    - Filing or commenting on issues on GitHub
+    - Contributing to SIG, subproject, or community discussions (e.g. meetings, Slack, email discussion
+      forums, Stack Overflow)
+- Actively contributing to 1 or more subprojects.
+- Sponsored by 2 approvers or leads. **Note the following requirements for sponsors**:
+    - Sponsors must have close interactions with the prospective member - e.g. code/design/proposal review, coordinating
+      on issues, etc.
+    - Sponsors must be approvers in at least 1 OWNERS file either in any of the four main subprojects in the [Argoproj org].
+      - An approver in the [Argoproj org] may sponsor someone for the [Argoproj org]
+      or any of the related [Argoproj GitHub organizations]; as long as it's a project they're involved with.
+      - A sponsor who is an approver in any of the related [Argoproj GitHub organizations]
+      but not in the [Argoproj org] can only sponsor someone for the orgs they are associated with.
+    - Sponsors must be from multiple member companies to demonstrate integration across community.
+- **[Open an issue][membership request] against the argoproj/argoproj repo**
+   - Ensure your sponsors are @mentioned on the issue
+   - Complete every item on the checklist ([preview the current version of the template][membership template])
+   - Make sure that the list of contributions included is representative of your work on the project.
+- Have your sponsoring reviewers reply confirmation of sponsorship: `+1`
+- Once your sponsors have responded, your request will be reviewed by the Argoproj GitHub Admin team. Any missing information will be requested.
+
+### Responsibilities and privileges
+
+- Responsive to issues and PRs assigned to them
+- Responsive to mentions of subprojects they are members of
+- Active owner of code they have contributed (unless ownership is explicitly transferred)
+  - Code is well tested
+  - Tests consistently pass
+  - Addresses bugs or issues discovered after code is accepted
+- Members are welcome and encouraged to review PRs and proposals.
+- They can be assigned to issues and PRs, and people can ask members for reviews with a `/cc @username`.
+
+**Note:** members who frequently contribute code are expected to proactively
+perform code reviews and work towards becoming a primary *reviewer* for the
+subproject that they are active in.
+
+
+
+## Reviewer
+
+Reviewers are able to review code for quality and correctness on some part of a
+subproject. They are knowledgeable about both the codebase and software
+engineering principles.
+
+**Defined by:** *reviewers* entry in an OWNERS file in a repo owned by the
+Argoproj project.
+
+Reviewer status is scoped to a part of the codebase.
+
+**Note:** Acceptance of code contributions requires at least one approver in
+addition to the assigned reviewers.
+
+### Requirements
```

Comment:
Could we add text to convey that these are non-binding guidelines for the maintainers who are going to approve new reviewers ?

Something like this:

```
While these guidelines are going to be used for setting contribution expectations, 
the maintainers will recognize the impact of an individual's contribution while 
making a decision on promotion an individual to a "reviewer" 
```

Reply:
added suggested item